presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -414,7 +414,12 @@ void collectDynamicFilters()
     private static Set<DynamicFilterId> getDynamicFiltersProducedInPlanNode(PlanNode planNode)
     {
         if (planNode instanceof JoinNode) {
-            return ((JoinNode) planNode).getDynamicFilters().keySet();
+            JoinNode joinNode = (JoinNode) planNode;
+            if (joinNode.isCrossJoin()) {
```

Comment:
Why would cross join have dynamic filters if there are no equi-conditions? Seems like a bug in planner.

Reply:
We'll be adding DFs for non-equi-joins  in https://github.com/prestosql/presto/pull/2674.

Reply:
Why cross join would produce DF. Currently, `io.prestosql.sql.planner.plan.JoinNode#isCrossJoin` is:
```
        return criteria.isEmpty() && filter.isEmpty() && type == INNER;
```
is it doesn't even contain non-equi joins.

Reply:
IIUC, in case of the query below, the join node has an empty `filter` and an empty `criteria`:
https://github.com/prestosql/presto/blob/77d87bf59fed989d841b026d884ce21932a49344/presto-main/src/test/java/io/prestosql/sql/planner/TestLocalDynamicFilterConsumer.java#L332-L344
So, I have used `isCrossJoin` in this PR, similarly to https://github.com/prestosql/presto/blob/6190bd11513321f9cf0fa5beb5ad52df6388cbe6/presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java#L1696-L1698
Please let me know if this should be implemented differently...

Reply:
@rzeyde-varada for inequality joins we rely that inequality conditions are pushed down to `JoinNode#filter` field. Joins that have neither equi or filter conditions won't filter anything.

see: `io/prestosql/sql/planner/LocalExecutionPlanner.java:2079` (part of `createLookupSourceFactory`)

Reply:
But if a join has only filter condition (e.g. [1]), it seems that the current predicate pushdown code (at `master`) doesn't push the `lineitem.orderkey <= orders.orderkey` into `JoinNode#filter` - so `LocalExecutionPlanner` is using `createNestedLoopJoin` instead of `createLookupJoin`.

Therefore, in #2674 we have added DF support for the "nested loop join" case.
I guess we should also add DF support for "lookup joins" with a non-empty filter.

[1]
```
SELECT * FROM lineitem, orders 
WHERE lineitem.orderkey <= orders.orderkey 
AND orders.comment = 'nstructions sleep furiously among '
```

Reply:
> But if a join has only filter condition (e.g. [1]), it seems that the current predicate pushdown code (at master) doesn't push the lineitem.orderkey <= orders.orderkey into JoinNode#filter - so LocalExecutionPlanner is using createNestedLoopJoin instead of createLookupJoin.

This seems like a bug. Could you figure why this happens? I'm pretty sure inequality expression should land in `JoinNode#filer`

Reply:
Sounds good, will do.

Reply:
I think it happens due to:
https://github.com/prestosql/presto/blob/80aa4354a88a54f4e74ea6685ba773df93cd8d94/presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java#L521-L528